components/collections/src/codepointtrie/cptrie.rs

```diff
@@ -438,6 +438,44 @@ impl<'trie, T: TrieValue> CodePointTrie<'trie, T> {
         })
     }
 
+    /// Maps the CodePointTrie into one that returns a different type.
+    ///
+    /// This function returns owned data.
+    ///
+    /// If the old and new types are the same size, use the more efficient
+    /// [`CodePointTrie::try_into_converted`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_collections::codepointtrie::CodePointTrie;
+    /// use icu_collections::codepointtrie::planes;
+    /// use core::convert::Infallible;
+    ///
+    /// let planes_trie_u8: CodePointTrie<u8> = planes::get_planes_trie();
+    /// let planes_trie_u16: CodePointTrie<u16> = planes_trie_u8
+    ///     .try_alloc_map_value(TryFrom::try_from)
+    ///     .expect("infallible");
+    ///
+    /// assert_eq!(planes_trie_u16.get32(0x30000), 3);
```

Comment:
totally optional. this is what i was thinking of in the previous comment. i wanted to show a x^3 function, but i wasn't sure what the Error type would look like for that.

```suggestion
    /// use icu_collections::codepointtrie::CodePointTrie;
    /// use icu_collections::codepointtrie::planes;
    /// use core::convert::Infallible;
    ///
    /// fn try_square(x: u8) -> Result<u16, Infallible> {
    ///   let y = x as u16;
    ///   Ok(y * y)
    /// }
    ///
    /// let planes_trie_u8: CodePointTrie<u8> = planes::get_planes_trie();
    /// let planes_trie_u16: CodePointTrie<u16> = planes_trie_u8
    ///     .try_alloc_map_value(try_square)
    ///     .expect("infallible");
    ///
    /// assert_eq!(planes_trie_u8.get32(0x10_0000), 16);
    /// assert_eq!(planes_trie_u16.get32(0x10_0000), 256);
```

Reply:
Yep; actually the square function could have an error condition in the case of overflow. It could be written with checked_mul.

I'm going to merge the PR as-is but may follow up with this.